File: 프로그래머스/unrated/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.js
function solution(arr, flag) {
    let result = [];
    for (let idx in arr) {
        if (flag[idx]) {
            Array.from({ length: arr[idx] * 2 }, () => result.push(arr[idx]));
        } else {
            result = result.slice(0, result.length - arr[idx]);
        }
    }
    return result;
}

File: 프로그래머스/unrated/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.java
import java.util.ArrayList;
class Solution {
    public int[] solution(int[] arr, boolean[] flag) {
        ArrayList< Integer > result = new ArrayList<>();
		for ( int i = 0; i < flag.length; i++ ) {
			if ( flag[ i ] ) {
				for ( int j = 0; j < arr[ i ] * 2; j++ ) {
					result.add( arr[ i ] );
				}
			} else {
				for ( int j = 0; j < arr[ i ]; j++ ) {
					result.remove( result.size() - 1 );
				}
			}
		}
		return result.stream().mapToInt( Integer :: intValue ).toArray();
    }
}